packages/flutter_tools/lib/src/isolated/native_assets/targets.dart

```diff
@@ -0,0 +1,378 @@
+      ),
+    ...dataAssetExtensions,
+  ];
+
+  @override
+  Future<void> setCCompilerConfig() async => cCompilerConfigSync = await cCompilerConfigMacOS();
+}
+
+final class AndroidAssetTarget extends CodeAssetTarget {
+  AndroidAssetTarget({
+    required super.architecture,
+    required Map<String, String> environmentDefines,
+    required super.supportedAssetTypes,
+  }) : _androidCodeConfig = AndroidCodeConfig(
+         targetNdkApi: targetAndroidNdkApi(environmentDefines),
+       ),
+       super(os: OS.android);
+
+  final AndroidCodeConfig? _androidCodeConfig;
+
+  @override
+  Future<void> setCCompilerConfig() async => cCompilerConfigSync = await cCompilerConfigAndroid();
+
+  @override
+  List<ProtocolExtension> get extensions => <ProtocolExtension>[
+    if (supportedAssetTypes.contains(SupportedAssetTypes.codeAssets))
+      CodeAssetExtension(
+        targetArchitecture: architecture,
+        linkModePreference: LinkModePreference.dynamic,
+        cCompiler: cCompilerConfigSync,
+        targetOS: OS.android,
+        android: _androidCodeConfig,
+      ),
+    ...dataAssetExtensions,
+  ];
+}
+
+final class FlutterTesterAssetTarget extends CodeAssetTarget {
+  FlutterTesterAssetTarget({required super.supportedAssetTypes})
+    : super(architecture: Architecture.current, os: OS.current) {
+    subtarget = switch (os) {
+      OS.linux => LinuxAssetTarget(
+        supportedAssetTypes: supportedAssetTypes,
+        architecture: architecture,
+      ),
+      OS.windows => WindowsAssetTarget(
+        supportedAssetTypes: supportedAssetTypes,
+        architecture: architecture,
+      ),
+      OS.macOS => MacOSAssetTarget(
+        supportedAssetTypes: supportedAssetTypes,
+        architecture: architecture,
+      ),
+      OS() => throw UnsupportedError('Flutter tester supports only linux, windows and macOS.'),
+    };
+  }
+
+  late final CodeAssetTarget subtarget;
+
+  @override
+  List<ProtocolExtension> get extensions => subtarget.extensions;
+
+  @override
+  CCompilerConfig? get cCompilerConfigSync => subtarget.cCompilerConfigSync;
+
+  @override
+  Future<void> setCCompilerConfig() async => subtarget.setCCompilerConfig();
+}
+
+List<AndroidArch> _androidArchs(TargetPlatform targetPlatform, String? androidArchsEnvironment) {
+  switch (targetPlatform) {
+    case TargetPlatform.android_arm:
+      return <AndroidArch>[AndroidArch.armeabi_v7a];
+    case TargetPlatform.android_arm64:
+      return <AndroidArch>[AndroidArch.arm64_v8a];
+    case TargetPlatform.android_x64:
+      return <AndroidArch>[AndroidArch.x86_64];
+    case TargetPlatform.android_x86:
+      return <AndroidArch>[AndroidArch.x86];
+    case TargetPlatform.android:
+      if (androidArchsEnvironment == null) {
+        throw MissingDefineException(kAndroidArchs, 'native_assets');
+      }
+      return androidArchsEnvironment.split(' ').map(getAndroidArchForName).toList();
+    case TargetPlatform.darwin:
```

Comment:
Nit: remove non-Android targets and use `default`.

Reply:
This would infringe on https://dart.dev/tools/linter-rules/no_default_cases